roles/populate_mirror_registry/tasks/prerequisites.yml

```diff
@@ -16,39 +16,73 @@
     - "{{ downloads_path }}"
     - "{{ downloads_path }}/{{ openshift_full_version }}"
 
+- name: Check if artifacts from release image have already been downloaded
+  stat:
+    path: "{{ downloads_path }}/{{ openshift_full_version }}/release.txt"
+  register: downloaded_artifacts
+
+# extract artifacts from the OCP release image instead of downloading them
+# from the release URL
+# This OCP release image provides the artifacts required (OCP clients, installer,
+# release.txt, among others)
+- name: Extract artifacts from release image
+  become: true
+  block:
+    # we need to temporarily download ocp stable binary to extract the artifacts
+    - name: Create tmp directory for oc stable client
+      tempfile:
+        state: directory
+        prefix: oc.
+      register: oc_tmp_dir
+
+    - name: Download stable client
+      unarchive:
+        src: "{{ stable_release_url }}/{{ stable_oc_tar }}"
+        dest: "{{ oc_tmp_dir.path }}"
+        mode: 0755
+        remote_src: yes
+
+    - name: Extract artifacts from release image (x86_64)
+      command:
+        cmd: >
+          "{{ oc_tmp_dir.path }}/oc" adm release extract
+            --registry-config "{{ config_file_path }}/{{ pull_secret_file_name }}"
+            --tools
+            --from "{{ release_image_item.remote | quote }}"
+            --to "{{ downloads_path }}/{{ openshift_full_version }}"
+      when: arm_release_image_key not in image_hashes
+
+    - name: Extract artifacts from release image (arm64)
+      command:
+        cmd: >
+          "{{ oc_tmp_dir.path }}/oc" release extract
+            --registry-config "{{ config_file_path }}/{{ pull_secret_file_name }}"
+            --tools
+            --from "{{ release_image_item.remote | quote }}"
+            --to "{{ downloads_path }}/{{ openshift_full_version }}"
+      when: arm_release_image_key in image_hashes
```

Comment:
Can you explain what this task is meant to do, it looks identical other than `adm` missing from  `/oc release extract`.

Reply:
Oh, you're right, I'm missing `adm`, I'll add it.
The command is working because I've tested it in x86 (the task before), by the way.
the explanation of the task is above the block that includes all this logic, mainly this:
```
# extract artifacts from the OCP release image instead of downloading them
# from the release URL
# This OCP release image provides the artifacts required (OCP clients, installer,
# release.txt, among others)
```
I hope this is fine, else I can write more comments, but it's mainly the same that we've included for our dci-openshift-agent.